includes/admin/class-sensei-tour.php

```diff
@@ -88,12 +88,86 @@ public function enqueue_admin_scripts( $hook ) {
 		 */
 		$tour_loaders = apply_filters( 'sensei_tour_loaders', $tour_loaders );
 
-		if ( ! empty( $tour_loaders ) ) {
+		$incomplete_tours = [];
+
+		foreach ( $tour_loaders as $handle => $tour_loader ) {
+			/**
+			 * Filters the tour completion status.
+			 *
+			 * @hook sensei_tour_is_complete Check if a tour is complete.
+			 *
+			 * @since $$next-version$$
+			 *
+			 * @param {bool}  $is_tour_complete The tour completion status.
+			 * @param {string} $tour_id The tour ID.
+			 *
+			 * @return {bool} Filtered tour completion status.
+			 */
+			$is_tour_complete = apply_filters( 'sensei_tour_is_complete', $this->get_tour_completion_status( $handle, get_current_user_id() ), $handle );
+			if ( ! $is_tour_complete ) {
+				$incomplete_tours[ $handle ] = $tour_loader;
+			}
+		}
+
+		if ( ! empty( $incomplete_tours ) ) {
 			Sensei()->assets->enqueue( 'sensei-tour-styles', 'admin/tour/style.css', [] );
 		}
 
-		foreach ( $tour_loaders as $handle => $tour_loader ) {
+		foreach ( $incomplete_tours as $handle => $tour_loader ) {
 			Sensei()->assets->enqueue( $handle, $tour_loader['path'], [], true );
 		}
 	}
+
+	/**
+	 * Set tour status for user.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $tour_id The tour ID.
+	 * @param bool   $status  The tour status.
+	 * @param int    $user_id The user ID.
+	 */
+	public function set_tour_completion_status( $tour_id, $status, $user_id = 0 ) {
+		$user_id = $user_id ? $user_id : get_current_user_id();
+
+		if ( ! $user_id ) {
+			return;
+		}
+
+		$tours = get_user_meta( $user_id, 'sensei_tours', true );
```

Comment:
Feel free to skip this one if you don't want to refactor it. ;)

Since this user meta is an array, I wonder if it would be cleaner and more standard if we used the `$single` as `false` instead, and we just add/remove what is needed from the meta array.

Reply:
Thanks! Yap that's a good alternative. I just preferred this way because in this way, we get a flexible data structure in case we want to add some extra fields or some additional data, we have easier control.